snuba/query/snql/__init__.py

```diff
@@ -29,19 +35,23 @@
     visit_parameters_list,
     visit_quoted_literal,
 )
-from typing import Sequence
-
 
 snql_grammar = Grammar(
     fr"""
-    query_exp             = match_clause where_clause? collect_clause? group_by_clause? having_clause? order_by_clause?
+    query_exp             = match_clause where_clause collect_clause group_by_clause? having_clause? order_by_clause?
 
     match_clause          = space* "MATCH" space* open_paren clause close_paren space*
-    where_clause          = space* "WHERE" clause space*
+    where_clause          = space* "WHERE" or_expression space*
     collect_clause        = space* "COLLECT" collect_list space*
     group_by_clause       = space* "BY" group_list space*
+    having_clause         = space* "HAVING" or_expression space*
     order_by_clause       = space* "ORDER BY" order_list space*
 
+    condition             = low_pri_arithmetic condition_op (column_name / numeric_literal)
```

Comment:
I think you also need string literals on the right hand side of this expression.
Also thinking over this, I think we will need to support functions on the right hand side as well. 
Think about `timestamp > toDateTime('2020-01-01')`
I don't like making the language silently convert the string type in the datetime. We can introduce that conversion later on and limit the allowed functions on the right hand side on after parsing.

A good idea here would be to ask for opinions on the SnQL discussion channel. You could ask people there what they think, if they prefer `timestamp > '2020-01-01'` like we do today or if they prefer to provide the function call explicitly. In the meantime you can accept functions.